remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java

```diff
@@ -95,17 +99,31 @@ public void onClick(View view) {
    */
 
   public void showRemixer(FragmentManager manager, String tag) {
-    if (!isAdded() && !isVisible() && !isRemoving()) {
-      show(manager, tag);
+    synchronized(syncLock) {
+      if (!isAddingFragment && !isAdded()) {
+        isAddingFragment = true;
+        show(manager, tag);
+      }
     }
   }
-  
+
+  // TODO(nicksahler): Generalize to attaching to any SensorEventListener
+
   /**
    * Attach this instance to a shake gesture and show fragment when magnitude exceeds {@code threshold}
    *
```

Comment:
in javadoc, no need to put that extra line there, also you can attach the todo to the Javadoc, or just below it, too.